src/flb_ra_key.c

```diff
@@ -145,7 +145,8 @@ static int subkey_to_object(msgpack_object *map, struct mk_list *subkeys,
             }
 
             /* Index limit */
-            if (cur.via.array.size < entry->array_id + 1) {
+            if (entry->array_id == 2147483647 ||
```

Comment:
what is ```2147483647``` ?, can you use a variable instead with proper identification of the value means ?

Reply:
Yes, thanks!

Reply:
How about using `INT_MAX` ?
It is defined at `<limits.h>`

Reply:
Great idea, how do you like it now?

Reply:
LGTM. Thank you.